udata/core/dataset/rdf.py

```diff
@@ -468,6 +469,31 @@ def title_from_rdf(rdf, url):
             return i18n._("Nameless resource")
 
 
+def primary_topic_identifier_from_rdf(rdf: RdfResource):
```

Comment:
WDYT ? https://github.com/opendatateam/udata/pull/3157/commits/c6c5c124afe86b10eb022b8f2d24605d04de2243

It forces a full graph re-parsing, but it's probably easier to understand what we're looking for (and thanks, because I actually had no idea 🙈). 

It's unclear to me in which case we pass a `node` to `dataset_from_rdf` and thus in which case fetching the first `CatalogRecord` from the graph could be a problem. The tests are OK 🤷.

Reply:
Well, after digging a bit, the tests work by chance (we probably always process the first item of the graph). I have to find a way to get the CatalogRecord associated to the Dataset in the graph.

Reply:
@maudetes https://github.com/opendatateam/udata/pull/3157/commits/f77473f9f158c6c93063f82e8511d6773c8bc995 should be generic enough and handle relationship both ways. WDYT?